camel-core/src/main/java/org/apache/camel/ComponentVerifier.java

```diff
@@ -21,118 +21,287 @@
+             * @return code name
+             */
+            String name();
         }
 
         /**
-         * @return the scope against which the parameters have been validated.
+         * Standard set of error codes
          */
-        Scope getScope();
+        enum StandardCode implements Code {
+            /**
+             * Authentication failed
+             */
+            AUTHENTICATION,
+            /**
+             * An exception occurred
+             */
+            EXCEPTION,
+            /**
+             * Internal error while performing the verification
+             */
+            INTERNAL,
+            /**
+             * A mandatory parameter is missing
+             */
+            MISSING_PARAMETER,
+            /**
+             * A given parameter is not known to the component
+             */
+            UNKNOWN_PARAMETER,
+            /**
+             * A given parameter is illegal
+             */
+            ILLEGAL_PARAMETER,
+            /**
+             * A combination of parameters is illegal. See {@link VerificationError#getParameterKeys()} for the set
+             * of affected parameters
+             */
+            ILLEGAL_PARAMETER_GROUP_COMBINATION,
+            /**
+             * A parameter <em>value</em> is not valid
+             */
+            ILLEGAL_PARAMETER_VALUE,
+            /**
+             * A group of parameters is not complete in order to be valid
+             */
+            INCOMPLETE_PARAMETER_GROUP,
+            /**
+             * The verification is not supported
+             */
+            UNSUPPORTED,
+            /**
+             * The requested {@link Scope} is not supported
+             */
+            UNSUPPORTED_SCOPE,
+            /**
+             * Generic error which is explained in more details with {@link VerificationError#getDetails()}
+             */
+            GENERIC
+        }
 
         /**
-         * @return the status
+         * Interface defining an attribute which is a key for the detailed error messages. This is implemented by several
+         * standard enums like {@link ExceptionAttribute}, {@link HttpAttribute} or {@link GroupAttribute} but can also
+         * implemented for component specific details. This is best done via {@link ResultErrorBuilder#asAttribute(String)}
+         * or using one of the other builder method in this error builder (like {@link ResultErrorBuilder#detail(String, Object)}
+         *
+         * With respecting to name, the same rules as for {@link Code} apply: Standard attributes are all upper case with _
+         * as separators, whereas custom attributes are lower case with underscore separators.
          */
-        Status getStatus();
+        interface Attribute {
+            /**
+             * @return Name of the attribute
+             */
+            String name();
+        }
```

Comment:
For the enum implementations this should be no problem.

For custom implementation, you are right because if you compare them you have probably two different instances. Maybe we should simply add a default `equals()` and `hashCode()` which defers to the one of `name()` ?